src/Microsoft.Azure.Functions.Extensions.Mcp/Trigger/McpToolListener.cs

```diff
@@ -26,6 +27,23 @@ internal sealed class McpToolListener(ITriggeredFunctionExecutor executor,
 
     public ICollection<IMcpToolProperty> Properties { get; set; } = properties;
 
+    public JsonElement? InputSchema 
+    { 
+        get => _inputSchema; 
+        set
+        {
+            if (value.HasValue && !McpInputSchemaJsonUtilities.IsValidMcpToolSchema(value.Value))
+            {
+                throw new ArgumentException(
```

Comment:
I know I had suggested something like this prior but I think its better we validate this schema where we read it and make this read only `public JsonElement? InputSchema => _inputSchema;`

So validation would actually happen on "GetInputSchema"

```csharp
        // Validate that the parsed schema is a valid MCP tool input schema
        if (!McpInputSchemaJsonUtilities.IsValidMcpToolSchema(schema))
        {
            throw new ArgumentException(
                "The specified document is not a valid MCP tool input JSON schema.",
                nameof(attribute.InputSchema));
        }

        return schema;
```